### PR TITLE
run yaml-lint  and pylint as github action

### PR DESCRIPTION
**Pull request by ericLemanissier**
related to

- conan-io/conan-center-index#7561
- conan-io/hooks#358
- conan-io/hooks/issues/308

This is not a library, but a github action job running yaml lint on yml files , and pylint on recipes of pull requests. A concrete example of failure of this job can be found on this PR (ags recipe), just as a demo that will be reverted before (hopefully) it is merged.

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
I added github actions annotations, so that the error are reported directly on the web interface of the pull requests:

- [yamllint](https://github.com/conan-io/conan-center-index/pull/7749/files#diff-31738eee15920d18d81977c61da865b7cd321061eef2877effd1abcf62b7beaf)
- [pylint](https://github.com/conan-io/conan-center-index/pull/7749/files#diff-186f3099de803923ae7baec393829188db4cd9e33b1c44746a9683affbea08c7)

What do you think ?

**Comment by ericLemanissier** [comment]
Call for reviews @uilianries @SSE4 @jgsogo @Croydon @Minimonium @SpaceIm @madebr @toge @danimtb @lasote 
Here are examples of real bugs which can be caught by the recipe-linter, and how they are shown on github:
image

image

image

image

image


they can be seen directly on this PR:
- [yamllint](https://github.com/conan-io/conan-center-index/pull/7749/files#diff-4ffe5d1b518bc1a0465eb141a97a84a8af0162d730dad882990ffee451eb75a9)
- [pylint](https://github.com/conan-io/conan-center-index/pull/7749/files#diff-a91894ef50b4c723f281df7076d46259739a858c5eefc9264639f4a2b86bb1c6)

**Review by SSE4 — DISMISSED** [review]
@ericLemanissier I really like it, it looks neat!

**Comment by uilianries** [comment]
I really liked this linter, but I'm afraid of Github actions time limit. So far, it offers 2k minutes/month (around 1h/day), but for the entire conan-io organization, and we have more projects running. I believe we could internalize it on Jenkins, where we have more resources.

**Comment by ericLemanissier** [comment]
The 2k limit applies only to private repos : https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions
Public repos have no such limits, but only restrictions on individual job/workflow length, and concurrent jobs : https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits

**Comment by ericLemanissier** [comment]
@uilianries did my message dissipate your concerns about github actions quota ?

**Comment by uilianries** [comment]
@ericLemanissier Thank you for explaining the details. Indeed your linter looks good and have my approval to run under Github Actions, so all community can collaborate and it will be transparent.

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by ericLemanissier** [comment]
ok, so I just removed the modifications to libpq which were just demonstrating the error reporting. Thanks for your reviews!

**Review by uilianries — DISMISSED** [review]
LGTM

Thank you @ericLemanissier !

File: .github/workflows/linters.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+      - name: install hook
+        run: |
+          conan config install https://github.com/conan-io/hooks.git
+          conan config set hooks.${{ matrix.hook }}
+
+      - name: run lint
+        run:  |
+          echo "::add-matcher::${{ matrix.hook }}.json"
+          python3 runlint.py ${{ github.event.pull_request.number }}
```
Comment (prince-chrismc) [inline .github/workflows/linters.yml:34]: on event `wokflow_dispatch` the PR number is still available? I dont see it in the docs https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#workflow_dispatch

I though an input value would be required 🤔

**Comment by prince-chrismc** [comment]
I have never heard/seen the [`add-matcher`](https://github.com/conan-io/conan-center-index/pull/7749/files#diff-c94729f5aa1afabfd250d3a929cdc840199d92facd4c83fcbc37b661d784d4acR33) before. It's really exciting! https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md Is super helpful!

My experience with the Annotations has been painful since the permission from public forks is not enough to post the results.

Does the problem matcher suffer from the same limitations?

File: recipe_linter.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (prince-chrismc) [inline recipe_linter.json:1]: Since these are specific to github and I like clean root directories, perhaps these can be moved to the `.github` folder?

**Comment by ericLemanissier** [comment]
Thanks for the feedback!

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by ericLemanissier** [comment]
wow, I thought a manual merge by an admin would be needed. thanks !
I'll keep an eye on these actions in the coming days, to be sure to react quickly, if this action breaks CI

**Pull request merged**